File: src/test/java/com/claudio_personagens/bookingcharactersapi/controllers/UserControllerIT.java
package com.claudio_personagens.bookingcharactersapi.controllers;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Profile;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.jdbc.Sql;
import org.testcontainers.junit.jupiter.Testcontainers;

import javax.transaction.Transactional;
import java.util.HashMap;
import static org.junit.Assert.assertEquals;

@Testcontainers
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@Transactional
public class UserControllerIT extends  AbstractControllerIT {

    @Test
    public void loginWithAdminUser() throws Exception {
        HashMap<String, String> login = getUser("admin", "123456");
        ResponseEntity<String> result  = testRestTemplate
                .postForEntity("/auth", new ObjectMapper().writeValueAsString(login), String.class);
        assertEquals(HttpStatus.UNAUTHORIZED, result.getStatusCode());
    }

    @Test
    public void loginWithUserNotAllowed() throws Exception {
        HashMap<String, String> login = getUser("admin-test", "password-test");

        ResponseEntity<String> result  = testRestTemplate
                .postForEntity("/auth", new ObjectMapper().writeValueAsString(login), String.class);
        assertEquals(HttpStatus.UNAUTHORIZED, result.getStatusCode());
    }

    private HashMap<String, String> getUser(String admin, String password) {
        HashMap<String, String> login = new HashMap<>();
        login.put("username", admin);
        login.put("password", password);
        return login;
    }
}

File: src/test/java/com/claudio_personagens/bookingcharactersapi/controllers/UserControllerIT.kt
package com.claudio_personagens.bookingcharactersapi.controllers

import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.Assert
import org.junit.Before
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.http.HttpStatus
import org.springframework.http.MediaType
import org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
import org.springframework.test.web.servlet.result.MockMvcResultHandlers
import org.springframework.test.web.servlet.result.MockMvcResultMatchers
import org.springframework.test.web.servlet.setup.DefaultMockMvcBuilder
import org.springframework.test.web.servlet.setup.MockMvcBuilders
import org.testcontainers.junit.jupiter.Testcontainers
import java.util.*
import javax.transaction.Transactional

@Testcontainers
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@Transactional
class UserControllerIT : AbstractControllerIT() {

    @Test
    fun loginWithAdminUser() {
        val login = getUser("anderson@test.com", "admin")
        this.mockMvc.perform(
            MockMvcRequestBuilders.post("/api/auth")
                .contentType(MediaType.APPLICATION_JSON)
                .content(ObjectMapper().writeValueAsString(login))
            )
            .andDo(MockMvcResultHandlers.print())
            .andExpect(MockMvcResultMatchers.status().isUnauthorized)
            .andReturn()
    }

    @Test
    fun loginWithUserNotAllowed() {
        val login = getUser("admin-test@test.com", "password")
        this.mockMvc.perform(
            MockMvcRequestBuilders.post("/api/auth")
                .contentType(MediaType.APPLICATION_JSON)
                .content(ObjectMapper().writeValueAsString(login))
            )
            .andDo(MockMvcResultHandlers.print())
            .andExpect(MockMvcResultMatchers.status().isUnauthorized)
            .andReturn()
    }

    private fun getUser(admin: String, password: String): HashMap<String, String> {
        val login = HashMap<String, String>()
        login["email"] = admin
        login["password"] = password
        return login
    }
}